internal/operators/cnv/cnv_operator.go

```diff
@@ -86,54 +86,30 @@ func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, ho
-		status := "All host roles must be assigned to enable OCS."
+		status := "All host roles must be assigned to enable CNV."
 		o.log.Info("Validate Requirements status ", status)
 		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetClusterValidationID(), Reasons: []string{status}}, nil
-	} else if host.Role == models.HostRoleWorker {
-		cpu, _ := o.GetCPURequirementForWorker(ctx, cluster)
-		if inventory.CPU.Count < cpu {
-			return api.ValidationResult{Status: api.Failure, ValidationId: o.GetClusterValidationID(), Reasons: []string{fmt.Sprintf("Insufficient CPU to deploy CNV. Required CPU count is %d but found %d ", cpu, inventory.CPU.Count)}}, nil
-		}
-		mem, _ := o.GetMemoryRequirementForWorker(ctx, cluster)
-		if inventory.Memory.UsableBytes < mem {
-			usableMemory := conversions.BytesToMib(inventory.Memory.UsableBytes)
-			memBytes := conversions.BytesToMib(mem)
-			return api.ValidationResult{Status: api.Failure, ValidationId: o.GetClusterValidationID(), Reasons: []string{fmt.Sprintf("Insufficient memory to deploy CNV. Required memory is %d MiB but found %d MiB", memBytes, usableMemory)}}, nil
-		}
-	} else if host.Role == models.HostRoleMaster {
-		// TODO: validate available devices on worker node like gpu and sr-iov and check whether there is enough memory to support them
-		cpu, _ := o.GetCPURequirementForMaster(ctx, cluster)
-		if inventory.CPU.Count < cpu {
-			return api.ValidationResult{Status: api.Failure, ValidationId: o.GetClusterValidationID(), Reasons: []string{fmt.Sprintf("Insufficient CPU to deploy CNV. Required CPU count is %d but found %d ", cpu, inventory.CPU.Count)}}, nil
-		}
-		mem, _ := o.GetMemoryRequirementForMaster(ctx, cluster)
-		if inventory.Memory.UsableBytes < mem {
-			usableMemory := conversions.BytesToMib(inventory.Memory.UsableBytes)
-			memBytes := conversions.BytesToMib(mem)
-			return api.ValidationResult{Status: api.Failure, ValidationId: o.GetClusterValidationID(), Reasons: []string{fmt.Sprintf("Insufficient memory to deploy CNV. Required memory is %d MiB but found %d MiB", memBytes, usableMemory)}}, nil
-		}
 	}
-	return api.ValidationResult{Status: api.Success, ValidationId: o.GetClusterValidationID()}, nil
-}
-
-// GetCPURequirementForWorker provides worker CPU requirements for the operator
-func (o *operator) GetCPURequirementForWorker(_ context.Context, _ *common.Cluster) (int64, error) {
-	return workerCPU, nil
-}
-
-// GetCPURequirementForMaster provides master CPU requirements for the operator
-func (o *operator) GetCPURequirementForMaster(_ context.Context, _ *common.Cluster) (int64, error) {
-	return masterCPU, nil
-}
+	requirements, err := o.GetHostRequirementsForRole(ctx, cluster, host.Role)
+	if err != nil {
+		message := fmt.Sprintf("Failed to get host requirements for host with id %s", host.ID)
+		o.log.Error(message)
+		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetClusterValidationID(), Reasons: []string{message}}, err
```

Comment:
What do you think about adding `err.Error()` to `Reasons`?

Reply:
OK, I think in this case it is safe to add it to `Reasons`. 

If there was a possibility to have the error coming from deep app internals or DB connection I would be against it, as there might be information in there that a user shouldn't see.